OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -23,6 +23,14 @@ namespace OpenRA.Mods.Common.Traits
 	public class GivesBuildableAreaInfo : TraitInfo<GivesBuildableArea> { }
 	public class GivesBuildableArea { }
 
+	public enum FootprintCellType
+	{
+		Empty = '_',
+		OccupiedPassable = '=',
+		Blocking = 'x',
+		BlockingUntargetable = 'u'
```

Comment:
How about `X` instead of `u`?  This would be more intuitive IMO as this is only a minor modification of the normal `x`.